testing/python/fixtures.py

```diff
@@ -3950,3 +3950,35 @@ def fix():
 
     with pytest.raises(pytest.fail.Exception):
         assert fix() == 1
+
+
+def test_fixture_param_shadowing(testdir):
+    """Parametrized arguments would be shadowed if a fixture with the same name also exists (#5036)"""
+    testdir.makepyfile(
+        """
+        import pytest
+
+        @pytest.fixture(params=['a', 'b'])
+        def argroot(request):
+            return request.param
+
+        @pytest.fixture
+        def arg(argroot):
+            return argroot
+            
+        # This should only be parametrized directly
+        @pytest.mark.parametrize("arg", [1])
+        def test_direct(arg):
+            assert arg == 1
+            
+        # This should be parametrized based on the fixtures
+        def test_indirect(arg):
+            assert isinstance(arg, str)
+
+    """
+    )
+    # Only one test should have run
+    result = testdir.runpytest("-v")
+    result.stdout.fnmatch_lines(["::test_direct[1]"])
```

Comment:
My bad, those should be:

```python
    result.stdout.fnmatch_lines(["*::test_direct[[]1[]]*"])
    result.stdout.fnmatch_lines(["*::test_indirect[[]a[]]*"])
    result.stdout.fnmatch_lines(["*::test_indirect[[]b[]]*"])
```

(In glob syntax, to escape `[` one needs to use `[[]`, and for `]` to use `[]]`; ugly I know 😷)

Reply:
Great, was about to search :)